self-hosted runner not connected to my repo.

Hello, I am new to cml community. I am trying to use self-hosted runners with github actions to set up CI with my private github repo. I have pulled the docker container dvcorg/cml-py3 and ran with all the environment variables as specified in this video https://www.youtube.com/watch?v=rVq-SCNyxVc&ab_channel=DVCorgDVCorg 

I have created a personal access token with workflow and repo permissions.

However, when I run this without -d flag, I see an error as follows:

```
{"level":"error","status":"terminated"}
Error: REPO_TOKEN does not have enough permissions to access workflow API
    at CML.repo_token_check (/cml/src/cml.js:236:13)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
	Destroy scheduled: 30 seconds remaining.
	No TF resource found
```
@elleobrien , Can you please take a look and help me understand, what I am doing wrong?

@0x2b3bfa0 ,after running the debug version as shown above, Please find the error below:

```
/usr/bin/cml-send-github-check -> /usr/lib/node_modules/@dvcorg/cml/bin/cml-send-github-check.js
/usr/bin/cml-send-comment -> /usr/lib/node_modules/@dvcorg/cml/bin/cml-send-comment.js
/usr/bin/cml-publish -> /usr/lib/node_modules/@dvcorg/cml/bin/cml-publish.js
/usr/bin/cml-runner -> /usr/lib/node_modules/@dvcorg/cml/bin/cml-runner.js
/usr/bin/cml-tensorboard-dev -> /usr/lib/node_modules/@dvcorg/cml/bin/cml-tensorboard-dev.js
/usr/bin/cml-cloud-runner-entrypoint -> /usr/lib/node_modules/@dvcorg/cml/bin/cml-runner.js
/usr/bin/cml-pr -> /usr/lib/node_modules/@dvcorg/cml/bin/cml-pr.js

> mmmagic@0.5.3 install /usr/lib/node_modules/@dvcorg/cml/node_modules/mmmagic
> node-gyp rebuild

make: Entering directory '/usr/lib/node_modules/@dvcorg/cml/node_modules/mmmagic/build'
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/apprentice.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/apptype.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/ascmagic.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/cdf.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/cdf_time.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/compress.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/der.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/encoding.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/fsmagic.o
../deps/libmagic/src/fsmagic.c: In function ‘file_fsmagic’:
../deps/libmagic/src/fsmagic.c:223:13: warning: In the GNU C Library, "major" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "major", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "major", you should undefine it after including <sys/types.h>.
        COMMA, (long)major(sb->st_rdev),
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                             
../deps/libmagic/src/fsmagic.c:224:13: warning: In the GNU C Library, "minor" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "minor", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "minor", you should undefine it after including <sys/types.h>.
        (long)minor(sb->st_rdev)) == -1)
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                             
../deps/libmagic/src/fsmagic.c:258:13: warning: In the GNU C Library, "major" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "major", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "major", you should undefine it after including <sys/types.h>.
        COMMA, (long)major(sb->st_rdev),
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                             
../deps/libmagic/src/fsmagic.c:259:13: warning: In the GNU C Library, "minor" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "minor", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "minor", you should undefine it after including <sys/types.h>.
        (long)minor(sb->st_rdev)) == -1)
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                             
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/funcs.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/is_tar.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/magic.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/print.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/readcdf.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/readelf.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/softmagic.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/fmtcheck.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/strlcat.o
  CC(target) Release/obj.target/libmagic/deps/libmagic/src/strlcpy.o
  AR(target) Release/obj.target/deps/libmagic/magic.a
  COPY Release/magic.a
  CXX(target) Release/obj.target/magic/src/binding.o
  SOLINK_MODULE(target) Release/obj.target/magic.node
  COPY Release/magic.node
make: Leaving directory '/usr/lib/node_modules/@dvcorg/cml/node_modules/mmmagic/build'
+ @dvcorg/cml@0.4.3
added 2 packages from 9 contributors and updated 141 packages in 14.776s
{"level":"error","status":"terminated"}
	Destroy scheduled: 30 seconds remaining.
RequestError [HttpError]: Not Found
    at /usr/lib/node_modules/@dvcorg/cml/node_modules/@octokit/request/dist-node/index.js:68:23
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Github.runner_token (/usr/lib/node_modules/@dvcorg/cml/src/drivers/github.js:122:11)
    at async CML.runner_token (/usr/lib/node_modules/@dvcorg/cml/src/cml.js:140:12)
    at async CML.repo_token_check (/usr/lib/node_modules/@dvcorg/cml/src/cml.js:223:5)
    at async run (/usr/lib/node_modules/@dvcorg/cml/bin/cml-runner.js:290:3) {
  status: 404,
  headers: {
    'access-control-allow-origin': '*',
    'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset',
    connection: 'close',
    'content-encoding': 'gzip',
    'content-security-policy': "default-src 'none'",
    'content-type': 'application/json; charset=utf-8',
    date: 'Wed, 26 May 2021 21:34:34 GMT',
    'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
    server: 'GitHub.com',
    'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
    'transfer-encoding': 'chunked',
    vary: 'Accept-Encoding, Accept, X-Requested-With',
    'x-accepted-oauth-scopes': 'repo',
    'x-content-type-options': 'nosniff',
    'x-frame-options': 'deny',
    'x-github-media-type': 'github.v3; format=json',
    'x-github-request-id': 'D740:7A31:3A32F17:72B3631:60AEBEEA',
    'x-oauth-scopes': 'repo, workflow',
    'x-ratelimit-limit': '5000',
    'x-ratelimit-remaining': '4992',
    'x-ratelimit-reset': '1622066794',
    'x-ratelimit-resource': 'core',
    'x-ratelimit-used': '8',
    'x-xss-protection': '0'
  },
  request: {
    method: 'POST',
    url: 'https://api.github.com/repos/mhs-achyut/example_cml.git/actions/runners/registration-token',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-core.js/3.4.0 Node.js/12.22.1 (linux; x64)',
      authorization: 'token [REDACTED]',
      'content-length': 0
    },
    request: { agent: [Agent], hook: [Function: bound bound register] }
  },
  documentation_url: 'https://docs.github.com/rest/reference/actions#create-a-registration-token-for-a-repository'
}
	No TF resource found

```

--- a/src/cml.js
+++ b/src/cml.js
@@ -74,7 +74,10 @@ class CML {
   constructor(opts = {}) {
     const { driver, repo, token } = opts;
 
-    this.repo = uri_no_trailing_slash(repo || git_remote_url()).replace(/\.git$/, '');
+    this.repo = uri_no_trailing_slash(repo || git_remote_url()).replace(
+      /\.git$/,
+      ''
+    );
     this.token = token || infer_token();
     this.driver = driver || infer_driver({ repo: this.repo });
   }

--- a/src/cml.js
+++ b/src/cml.js
@@ -32,7 +32,7 @@ const git_remote_url = (opts = {}) => {
   const url = execSync(`git config --get remote.${remote}.url`).toString(
     'utf8'
   );
-  return strip_auth(git_url_parse(url).toString('https').replace('.git', ''));
+  return strip_auth(git_url_parse(url).toString('https'));
 };
 
 const infer_token = () => {
@@ -74,7 +74,7 @@ class CML {
   constructor(opts = {}) {
     const { driver, repo, token } = opts;
 
-    this.repo = uri_no_trailing_slash(repo || git_remote_url());
+    this.repo = uri_no_trailing_slash(repo || git_remote_url()).replace(/\.git$/, '');
     this.token = token || infer_token();
     this.driver = driver || infer_driver({ repo: this.repo });
   }

